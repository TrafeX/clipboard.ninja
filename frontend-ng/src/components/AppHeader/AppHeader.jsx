import {
  Box,
  Flex,
  Heading,
  HStack,
  IconButton,
  Image,
  Link,
  useColorMode,
  useColorModeValue
} from "@chakra-ui/react";
import NextImage from "next/image";
import NextLink from "next/link";
import { useRouter } from "next/router";
import { IoHelp } from "react-icons/io5";
import { MdDarkMode, MdLightMode } from "react-icons/md";
import ninjalogo from "./ninja.svg";

const AppHeader = () => {
  const { colorMode, toggleColorMode } = useColorMode();
  const textColor = useColorModeValue("white", "gray.800");
  const router = useRouter();

  return (
    <>
      <Box
        bg={"brand.500"}
        px={4}
        position={"sticky"}
        top={0}
        boxShadow={"md"}
        zIndex={2}
      >
        <Flex h={16} alignItems={"center"} justifyContent={"space-between"}>
          <HStack alignItems={"center"} spacing={1}>
            <Image
              src={ninjalogo}
              alt="Clipboard.ninja"
              height="50px"
              width="50px"
              as={NextImage}
            />
            <Heading
              as={NextLink}
              href={"/"}
              color={textColor}
              fontWeight={"normal"}
              size={"md"}
            >
              Clipboard.ninja
            </Heading>
          </HStack>
          <HStack alignItems={"center"} spacing={2}>
            {router.asPath !== "/" && (
              <Link as={NextLink} color={textColor} href={"/"}>
                Go back
              </Link>
            )}

            <NextLink href={router.asPath === "/about" ? "/" : "/about"}>
              <IconButton
                bg={"gray.100"}
                color={"gray.800"}
                _hover={{
                  bg: "gray.300",
                }}
                size={"sm"}
                icon={<IoHelp />}
                title={"Go to help"}
                aria-label={"Go to help"}
              />
            </NextLink>
            <IconButton
              bg={"gray.100"}
              color={"gray.800"}
              _hover={{
                bg: "gray.300",
              }}
              size={"sm"}
              icon={colorMode === "light" ? <MdDarkMode /> : <MdLightMode />}
              title={"Change Color Theme"}
              aria-label={"Change Color Theme"}
              onClick={toggleColorMode}
            />
          </HStack>
        </Flex>
      </Box>
      <Box
        position={"absolute"}
        top={0}
        left={0}
        height="240px"
        bgColor={"gray.400"}
        width={"100%"}
        zIndex={-1}
      />
    </>
  );
};

export default AppHeader;
