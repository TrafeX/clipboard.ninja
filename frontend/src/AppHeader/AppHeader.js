// @flow
import React from 'react';
import {AppBar, FlatButton} from 'material-ui';
import { withRouter } from 'react-router-dom';
import ninjalogo from './ninja.svg';

type Props = {
  history: Object,
  location: Location,
};
const AppHeader = (props: Props) => {
  let navButton = <FlatButton label="Help" href="/about.html" onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
    e.preventDefault();
    props.history.push('/about.html');
  }}/>;
  if (props.location.pathname === '/about.html') {
    navButton = <FlatButton label="Back" href="/" onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
      e.preventDefault();
      props.history.push('/');
    }}/>;
  }
  return (
    <div>
      <AppBar
        title="Clipboard.ninja"
        iconElementLeft={<img src={ninjalogo} alt="logo" height="50px"/>}
        iconElementRight={navButton}
        onTitleClick={() => props.history.push('/')}
      />
    </div>
  );
};

export default withRouter(AppHeader);

