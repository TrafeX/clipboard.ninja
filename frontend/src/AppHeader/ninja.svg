<svg xmlns="http://www.w3.org/2000/svg" width="207.873" height="264.567" viewBox="0 0 207.87334 264.56668"><defs><clipPath id="a"><path d="M0 198.425h155.905V0H0z"/></clipPath></defs><g clip-path="url(#a)" transform="matrix(1.33333 0 0 -1.33333 0 264.567)"><path d="M110.18 78.934s-8.27 17.048-13.37 19.78c-.14.074-.294.133-.437.202-5.962-3.022-12.853-4.375-19.988-3.51-2.627.317-5.145.927-7.538 1.78-2.372-1.335-3.57-2.567-3.57-2.567C54.89 84.233 49.623 78.294 49.623 78.294s-9.56-3.192-14.803-3.97c-2.504-.372-.315-1.55-.315-1.55s13.042-.705 16.778.747c3.39 1.32 15.8 10.365 16.26 10.72.327-.674 3.305-6.995 3.03-13.315-.326-7.478-4.117-13.852-4.117-13.852.64-.465 2.728-1.053 3.795-1.328 4.77-1.312 12.937-2.292 12.937-2.292l-.01-1.27s2.485.04 5.742.412c8.16.934 13.255 3.106 13.255 3.106C96.925 63.203 96 70.298 96 70.298c.054 4.194 1.444 8.457 3.343 12.328 2.083-2.23 5.43-5.692 7.117-6.776 2.552-1.64 13.543-5.676 13.543-5.676l.272 1.587c-4.263 1.007-10.096 7.174-10.096 7.174" fill="#0c363a"/><path d="M109.665 37.934c-.237 1.92-4.716 10.93-8.428 17.41-1.822-.657-6.212-2.05-12.318-2.747-3.258-.373-5.742-.413-5.742-.413l.008 1.27s-8.166.98-12.935 2.293c-.805.208-2.19.593-3.098.97-3.834-6.503-8.97-16.723-9.224-18.783-.082-.667.25-14.5-.886-23.17l1.415.138s7.47 17.032 6.12 19.55c0 0 19.377 15.518 19.585 15.514 3.925-.494 18.853-15.514 18.853-15.514-1.35-2.518 6.12-19.55 6.12-19.55l1.416-.137c-1.135 8.67-.803 22.502-.885 23.17" fill="#0c363a"/><path d="M82.64 68.73c-4.86-.248-8.247.685-10.408 1.64-.677.3-1.218.598-1.656.868-.002-.104 0-.207-.003-.31-.04-.942-.102-1.874-.27-2.76 2.506-1.267 6.553-2.582 12.483-2.27.083.007.168.01.252.015 6.27.323 10.225 1.3 13.25 2.912-.227.914-.288 1.473-.288 1.473.008.572.047 1.143.102 1.715-2.72-1.793-6.25-2.887-13.462-3.283" fill="#191516"/><path d="M79.335 65.968s.047-.905 1.865-.963c1.82-.058 2.76.242 3.235 1.83.473 1.588-1.106 2.87-1.988 2.37-.882-.5-1.398-.274-2.218-.013-.822.26-1.59-1.495-.895-3.224" fill="#191516"/><path d="M82.25 66.168s3.997-1.206 6.22-6.187c0 0 2.603.388 3.708 1.454 0 0-3.686 3.996-9.498 5.21zM83.21 68.257s-7.418-6.27-7.742-10.06c0 0-1.792 1.726-2.195 3.035 0 0 5.17 5.377 9.938 7.025" fill="#191516"/><path d="M58.775 15.632s-.497-1.38.146-2.762c.932-2.004-.486-1.997-2.135-2.537C53.323 9.2 50.28 5.696 47.27 4.77c-2.14-.66-2.51.027-2.777.694-.268.668.974 2.356.23 3.373-1.52 2.084 7.123 3.9 8.725 4.3 3.132.783 3.38.208 3.78 3.01.4 2.804 1.547-.515 1.547-.515" fill="#0c363a"/><path d="M109.665 37.934c-.257 2.08-5.5 12.5-9.346 18.99-1.33-.456-3.31-.982-4.118-1.177-5.09-1.312-9.934-1.59-9.934-1.59h-4.943s-4.845.278-9.934 1.59c-.808.195-2.788.72-4.117 1.177-3.845-6.49-9.088-16.91-9.345-18.99-.082-.667.25-14.5-.886-23.17l1.415.138s7.47 17.032 6.12 19.55c0 0 19.377 15.518 19.585 15.514 3.925-.494 18.853-15.514 18.853-15.514-1.35-2.518 6.12-19.55 6.12-19.55l1.416-.137c-1.135 8.67-.803 22.502-.885 23.17" fill="#0c363a"/><path d="M108.817 15.632s.497-1.38-.146-2.762c-.93-2.004.488-1.997 2.137-2.537 3.462-1.132 6.506-4.637 9.515-5.562 2.14-.66 2.51.027 2.777.694.267.668-.975 2.356-.232 3.373 1.52 2.084-7.122 3.9-8.724 4.3-3.132.783-3.38.208-3.78 3.01-.4 2.804-1.547-.515-1.547-.515" fill="#0c363a"/><path d="M82.64 68.73c-4.86-.248-8.247.685-10.408 1.64-.677.3-1.218.598-1.656.868-.002-.104 0-.207-.003-.31-.04-.942-.102-1.874-.27-2.76 2.506-1.267 6.553-2.582 12.483-2.27.083.007.168.01.252.015 6.27.323 10.225 1.3 13.25 2.912-.227.914-.288 1.473-.288 1.473.008.572.047 1.143.102 1.715-2.72-1.793-6.25-2.887-13.462-3.283" fill="#191516"/><path d="M79.335 65.968s.047-.905 1.865-.963c1.82-.058 2.76.242 3.235 1.83.473 1.588-1.106 2.87-1.988 2.37-.882-.5-1.398-.274-2.218-.013-.822.26-1.59-1.495-.895-3.224" fill="#191516"/><path d="M82.25 66.168s3.997-1.206 6.22-6.187c0 0 2.603.388 3.708 1.454 0 0-3.686 3.996-9.498 5.21zM83.21 68.257s-7.418-6.27-7.742-10.06c0 0-1.792 1.726-2.195 3.035 0 0 5.17 5.377 9.938 7.025" fill="#191516"/><path d="M58.775 15.632s-.497-1.38.146-2.762c.932-2.004-.486-1.997-2.135-2.537C53.323 9.2 50.28 5.696 47.27 4.77c-2.14-.66-2.51.027-2.777.694-.268.668.974 2.356.23 3.373-1.52 2.084 7.123 3.9 8.725 4.3 3.132.783 3.38.208 3.78 3.01.4 2.804 1.547-.515 1.547-.515" fill="#0c363a"/><path d="M109.665 37.934c-.257 2.08-5.5 12.5-9.346 18.99-1.33-.456-3.31-.982-4.118-1.177-5.09-1.312-9.934-1.59-9.934-1.59h-4.943s-4.845.278-9.934 1.59c-.808.195-2.788.72-4.117 1.177-3.845-6.49-9.088-16.91-9.345-18.99-.082-.667.25-14.5-.886-23.17l1.415.138s7.47 17.032 6.12 19.55c0 0 19.377 15.518 19.585 15.514 3.925-.494 18.853-15.514 18.853-15.514-1.35-2.518 6.12-19.55 6.12-19.55l1.416-.137c-1.135 8.67-.803 22.502-.885 23.17" fill="#0c363a"/><path d="M108.817 15.632s.497-1.38-.146-2.762c-.93-2.004.488-1.997 2.137-2.537 3.462-1.132 6.506-4.637 9.515-5.562 2.14-.66 2.51.027 2.777.694.267.668-.975 2.356-.232 3.373 1.52 2.084-7.122 3.9-8.724 4.3-3.132.783-3.38.208-3.78 3.01-.4 2.804-1.547-.515-1.547-.515" fill="#0c363a"/><path d="M30.428 81.292s-24.116 49.022-17.85 112.63l-5.42-4.38S1.77 133.53 25.183 80.15z" fill="#82c9d0"/><path d="M9.592 191.453l2.982 2.41c0 .02.002.038.005.06l-5.42-4.382S1.77 133.53 25.18 80.15l2.493.543C4.15 134.856 9.592 191.453 9.592 191.453" fill="#bde0e4"/><path d="M35.1 82.87l-7.723-5.006-6.78.5-.12.707 6.734 5.732 7.743-1.144z" fill="#191516"/><path d="M27.49 77.166l-6.9 1.206 6.733 5.73 7.77-1.158c.237-.074-7.603-5.778-7.603-5.778" fill="#3c3c3c"/><path d="M67.534 84.263c.124-.253 1.482-3.07 2.35-6.803 1.027 4.405-.02 8.13-.02 8.13s-.796-.61-2.33-1.327" fill="#001c26"/><path d="M82.64 68.73c-4.86-.248-8.247.685-10.408 1.64-.677.3-1.218.598-1.656.868-.002-.104 0-.207-.003-.31-.04-.942-.102-1.874-.27-2.76 2.506-1.267 6.553-2.582 12.483-2.27.083.007.168.01.252.015 6.27.323 10.225 1.3 13.25 2.912-.227.914-.288 1.473-.288 1.473.008.572.047 1.143.102 1.715-2.72-1.793-6.25-2.887-13.462-3.283" fill="#191516"/><path d="M79.335 65.968s.047-.905 1.865-.963c1.82-.058 2.76.242 3.235 1.83.473 1.588-1.106 2.87-1.988 2.37-.882-.5-1.398-.274-2.218-.013-.822.26-1.59-1.495-.895-3.224" fill="#191516"/><path d="M82.25 66.168s3.997-1.206 6.22-6.187c0 0 2.603.388 3.708 1.454 0 0-3.686 3.996-9.498 5.21zM83.21 68.257s-7.418-6.27-7.742-10.06c0 0-1.792 1.726-2.195 3.035 0 0 5.17 5.377 9.938 7.025" fill="#191516"/><path d="M36.955 65.774c.185-.58-1.383-1.597-3.502-2.27-2.118-.677-3.983-.753-4.168-.174-.185.58 1.382 1.596 3.5 2.27 2.118.676 3.986.753 4.17.174" fill="#3c3c3c"/><path d="M27.693 81.63c-1.634-.52-1.7-1.09-1.7-1.09l5.134-16.082h.002c.083-.236.962-.17 1.97.152 1.01.32 1.765.776 1.697 1.018h.002L29.664 81.71s-.34.44-1.972-.08" fill="#191516"/><path d="M125.477 80.713s24.116 49.022 17.85 112.63l5.42-4.38s5.39-56.012-18.025-109.39z" fill="#bde0e4"/><path d="M146.315 190.874l-2.983 2.41c-.003.02-.002.04-.006.06l5.42-4.382s5.39-56.01-18.025-109.39l-2.49.542c23.525 54.164 18.085 110.76 18.085 110.76" fill="#82c9d0"/><path d="M134.27 76.325l-9.174.762-5.06 4.536.337.634 8.833.42 5.427-5.636z" fill="#191516"/><path d="M124.76 76.464l-4.725 5.172 8.83.42 5.443-5.666c.14-.204-9.55.074-9.55.074M125.286 61.66c-.207-.57-2.07-.417-4.16.345-2.087.76-3.61 1.84-3.403 2.41.207.573 2.068.42 4.158-.34 2.09-.762 3.615-1.842 3.406-2.414" fill="#3c3c3c"/><path d="M127.648 79.872c-1.612.587-2.012.178-2.012.178l-5.767-15.867h.002c-.08-.24.658-.725 1.653-1.088.994-.362 1.87-.463 1.964-.23v-.002l5.768 15.866s0 .556-1.61 1.142" fill="#191516"/><path d="M34.863 74.337s-.908-.193-1.263.832c-.354 1.024-1.046 5.43-2.535 5.188-1.97-.318-2.704-.154-3.43-.6-.834-.512.01-1.925 3.248-1.872l2.305-7.215s1.053 1.4 1.317 2.104c.262.7.358 1.563.358 1.563M119.694 71.75s.9-.226 1.288.79c.388 1.01 1.22 5.395 2.7 5.104 1.96-.38 2.7-.24 3.41-.71.816-.538-.02-1.966-3.253-1.81l-2.57-7.102s-1.026 1.442-1.267 2.152c-.243.71-.31 1.575-.31 1.575" fill="#0c363a"/><path d="M26.398 75.18s-.91 1.265.262 1.878c.735.385 1.728.656 2.097-.46.37-1.118.045-1.735 1.166-3.72.942-1.666 1.065-3.06-.154-3.412-2.25-.645-2.628 4.895-3.372 5.715M128.325 72.622s.81 1.33-.405 1.85c-.764.328-2.04.606-2.32-.537-.284-1.14.086-1.733-.88-3.795-.812-1.737-.97-3.116.27-3.37 2.294-.472 2.654 4.977 3.335 5.852M58.775 15.632s-.497-1.38.146-2.762c.932-2.004-.486-1.997-2.135-2.537C53.323 9.2 50.28 5.696 47.27 4.77c-2.14-.66-2.51.027-2.777.694-.268.668.974 2.356.23 3.373-1.52 2.084 7.123 3.9 8.725 4.3 3.132.783 3.38.208 3.78 3.01.4 2.804 1.547-.515 1.547-.515M108.817 15.632s.497-1.38-.146-2.762c-.93-2.004.488-1.997 2.137-2.537 3.462-1.132 6.506-4.637 9.515-5.562 2.14-.66 2.51.027 2.777.694.267.668-.975 2.356-.232 3.373 1.52 2.084-7.122 3.9-8.724 4.3-3.132.783-3.38.208-3.78 3.01-.4 2.804-1.547-.515-1.547-.515M115.307 125.92c-2.32-19.175-19.747-32.837-38.922-30.515-19.175 2.322-32.837 19.748-30.515 38.922 2.322 19.174 19.748 32.836 38.92 30.516 19.177-2.322 32.84-19.748 30.517-38.923" fill="#0c363a"/><path d="M76.768 127.153s8.872-.03 17.53 2.07c8.66 2.1 12.125-2.82 8.774-11.024-3.355-8.208-13.198-12.664-17.033-11.59-4.692 1.318-7.727 4.218-10.42 2.893-2.694-1.326-6.545-4.127-12.51-.508-5.965 3.62-9.59 14.668-8.024 18.717 1.564 4.046 8.084-.416 21.682-.56" fill="#ffc275"/><path d="M56.687 126.63c1.564 4.047 8.084-.416 21.683-.56 0 0 8.87-.027 17.527 2.073 2.903.703 5.208.602 6.882-.13-1.628 1.642-4.496 2.178-8.483 1.21-8.657-2.1-17.53-2.07-17.53-2.07-13.597.143-20.117 4.606-21.68.56-1.567-4.05 2.058-15.1 8.024-18.718.137-.082.265-.144.4-.22-5.237 4.262-8.28 14.085-6.823 17.855" fill="#d1945e"/><path d="M56.723 151.032s9.3 10.94 20.24 10.777c0 0 2.377.275 2.698-1.674.325-1.95-2.236-2.217-4.035-2.518-1.432-.238-11.13-.94-18.902-6.586M84.632 159.132c-.66-.787-1.834-.89-2.62-.23-.787.662-.89 1.835-.23 2.623.662.786 1.836.888 2.622.226.788-.658.89-1.832.228-2.618" fill="#214a5a"/><path d="M71.46 121.066l5.825-2.542 10.064 5.243-10.016-3.66zM75.13 117.024c-.222-.747-.905-1.295-1.725-1.295-.8 0-1.472.523-1.71 1.245-.062-.185-.104-.38-.104-.588 0-1.01.82-1.83 1.83-1.83 1.01 0 1.83.82 1.83 1.83 0 .225-.046.438-.12.637M84.17 116.993c-.244-.825-1-1.432-1.906-1.432-.886 0-1.628.58-1.89 1.38-.07-.205-.116-.422-.116-.65 0-1.117.906-2.023 2.022-2.023 1.118 0 2.024.906 2.024 2.023 0 .248-.05.484-.133.703" fill="#191516"/><path d="M87.19 50.485c-.473-.672-1.292-1.02-1.292-1.02.355.582.647 1.437.647 1.437-.834-.77-1.44-.792-1.44-.792.508.94.538 1.46.538 1.46-.47-.585-1.02-1.155-1.48-1.606.566-.07 1.365-.452 2.31-1.038.44.844.718 1.56.718 1.56M63.483 36.405s.646-.785 1.128-1.928l1.907 1.526c-.85.22-1.962.43-3.034.402M102.94 34.525c.568 1.18 1.404 1.658 1.404 1.658-.89-.23-1.545-.614-2.03-1.037l.626-.62M101.237 55.343c-1.822-.656-6.212-2.048-12.318-2.746-2.37-.272-4.328-.366-5.22-.4l.023-.596s12.473.38 18.01 2.868c-.166.295-.333.59-.496.875m-34.085 1.374c-.126-.214-.257-.44-.386-.663 2.34-.697 10.284-2.947 16.415-3.244l.006.644s-8.166.98-12.935 2.293c-.805.207-2.19.593-3.098.97" fill="#001c26"/><path d="M83.103 53.464c.153 3.418.346 8.564.346 8.564-.596-2.88-1.585-6.232-2.236-8.316.98-.135 1.668-.22 1.89-.248M44.842 7.235c.792.943 1.794 1.785.92 1.697-.585-.058-.965.096-1.197.257.024-.117.074-.234.16-.353.33-.452.263-1.034.117-1.602m12.197 3.953c-1.505.226-2.18.15-3.61-1.69-1.43-1.843-6.353-4.4-7.743-4.475-.398-.02-.728-.01-.994.037.3-.488.88-.812 2.578-.29 3.01.926 6.053 4.43 9.515 5.563 1.125.368 2.122.496 2.36 1.166-.423-.42-1.102-.462-2.106-.312M122.702 7.433c-.1.455-.104.904.108 1.28-.618.06-1.22.407-1.03-.192.144-.444.555-.755.922-1.087m-2.8-1.732c-2.332.79-4.625 2.895-5.377 4.212-.752 1.316-1.842 1.503-3.307.977-.522-.188-1.065-.234-1.55-.22.345-.104.734-.204 1.14-.337 3.46-1.13 6.505-4.636 9.514-5.562 2.142-.66 2.51.027 2.777.694.005.016 0 .037.006.053-.41-.615-1.3-.46-3.205.184M81.885 91.905c6.688-.81 13.155.338 18.852 2.973-1.385 1.787-2.743 3.2-3.928 3.836-.14.074-.294.132-.437.202-5.962-3.023-12.853-4.375-19.988-3.51-2.627.316-5.146.927-7.538 1.778-.298-.168-.577-.334-.838-.496 4.11-2.486 8.796-4.17 13.875-4.783M98.177 85.154c-.497-1.958-.264-3.747.062-4.993.334.815.692 1.62 1.077 2.408-.914 1.583-1.14 2.586-1.14 2.586M49.622 78.295s-.59-.197-1.554-.504c.33-1.01 1.012-1.457 1.012-1.457.05.867.354 1.58.652 2.086l-.11-.125M110.182 78.93c-.08-.523-.252-1.12-.623-1.653 0 0 .96-.062 1.778.505-.708.678-1.146 1.14-1.156 1.15" fill="#001c26"/></g></svg>