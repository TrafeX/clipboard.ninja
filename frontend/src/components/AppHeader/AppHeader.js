// @flow
import React from 'react';
import {AppBar, FlatButton} from 'material-ui';
import { withRouter } from 'react-router-dom';
import ninjalogo from './ninja.svg';

type Props = {
  history: Object,
  location: Location,
};
const AppHeader = ({history, location}: Props) => {
  let navButton = <FlatButton label="Help" href="/about.html" onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
    e.preventDefault();
    history.push('/about.html');
  }}/>;
  if (location.pathname === '/about.html') {
    navButton = <FlatButton label="Back" href="/" onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
      e.preventDefault();
      history.push('/');
    }}/>;
  }
  return (
    <div>
      <AppBar
        title="Clipboard.ninja"
        iconElementLeft={<img src={ninjalogo} alt="logo" height="50px"/>}
        iconElementRight={navButton}
        onTitleClick={() => history.push('/')}
      />
    </div>
  );
};

export default withRouter(AppHeader);

