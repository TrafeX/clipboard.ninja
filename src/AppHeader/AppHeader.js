// @flow
import React, { Component } from 'react';
import {AppBar, FlatButton} from 'material-ui';
import { withRouter } from 'react-router-dom';
import ninjalogo from './ninja.svg';

type Props = {
  history: Object,
  location: Location,
};
class AppHeader extends Component<Props> {
  render() {
    let navButton = <FlatButton label="Help" href="/about.html" onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
      e.preventDefault();
      this.props.history.push('/about.html');
    }} />;
    if (this.props.location.pathname === '/about.html') {
      navButton = <FlatButton label="Back" href="/" onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
        e.preventDefault();
        this.props.history.push('/');
      }} />;
    }
    return (
      <div>
        <AppBar
          title="Clipboard.ninja"
          iconElementLeft={<img src={ninjalogo} alt="logo" height="50px" />}
          iconElementRight={navButton}
          onTitleClick={() => this.props.history.push('/')}
        />
      </div>
    );
  }
}

export default withRouter(AppHeader);

