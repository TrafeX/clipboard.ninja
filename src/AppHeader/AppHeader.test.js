import React from 'react';
import ReactDOM from 'react-dom';
import AppHeader from './AppHeader';

it.skip('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <AppHeader />
    ,div);
});
