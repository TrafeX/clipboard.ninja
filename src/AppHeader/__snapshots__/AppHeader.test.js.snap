// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with a link to Help 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppHeader
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "pathname": "/",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <AppBar
        iconElementLeft={
          <img
            alt="logo"
            height="50px"
            src="ninja.svg"
          />
        }
        iconElementRight={
          <FlatButton
            disabled={false}
            fullWidth={false}
            href="/about.html"
            label="Help"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          />
        }
        onTitleClick={[Function]}
        showMenuIconButton={true}
        title="Clipboard.ninja"
        zDepth={1}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "iconElementLeft": <img
          alt="logo"
          height="50px"
          src="ninja.svg"
        />,
        "iconElementRight": <FlatButton
          disabled={false}
          fullWidth={false}
          href="/about.html"
          label="Help"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={false}
        />,
        "onTitleClick": [Function],
        "showMenuIconButton": true,
        "title": "Clipboard.ninja",
        "zDepth": 1,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AppBar
          iconElementLeft={
            <img
              alt="logo"
              height="50px"
              src="ninja.svg"
            />
          }
          iconElementRight={
            <FlatButton
              disabled={false}
              fullWidth={false}
              href="/about.html"
              label="Help"
              labelPosition="after"
              labelStyle={Object {}}
              onClick={[Function]}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            />
          }
          onTitleClick={[Function]}
          showMenuIconButton={true}
          title="Clipboard.ninja"
          zDepth={1}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconElementLeft": <img
            alt="logo"
            height="50px"
            src="ninja.svg"
          />,
          "iconElementRight": <FlatButton
            disabled={false}
            fullWidth={false}
            href="/about.html"
            label="Help"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          />,
          "onTitleClick": [Function],
          "showMenuIconButton": true,
          "title": "Clipboard.ninja",
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders with a link to Home 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppHeader
    location={
      Object {
        "pathname": "/about.html",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <AppBar
        iconElementLeft={
          <img
            alt="logo"
            height="50px"
            src="ninja.svg"
          />
        }
        iconElementRight={
          <FlatButton
            disabled={false}
            fullWidth={false}
            href="/"
            label="Back"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          />
        }
        onTitleClick={[Function]}
        showMenuIconButton={true}
        title="Clipboard.ninja"
        zDepth={1}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "iconElementLeft": <img
          alt="logo"
          height="50px"
          src="ninja.svg"
        />,
        "iconElementRight": <FlatButton
          disabled={false}
          fullWidth={false}
          href="/"
          label="Back"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={false}
        />,
        "onTitleClick": [Function],
        "showMenuIconButton": true,
        "title": "Clipboard.ninja",
        "zDepth": 1,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AppBar
          iconElementLeft={
            <img
              alt="logo"
              height="50px"
              src="ninja.svg"
            />
          }
          iconElementRight={
            <FlatButton
              disabled={false}
              fullWidth={false}
              href="/"
              label="Back"
              labelPosition="after"
              labelStyle={Object {}}
              onClick={[Function]}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            />
          }
          onTitleClick={[Function]}
          showMenuIconButton={true}
          title="Clipboard.ninja"
          zDepth={1}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconElementLeft": <img
            alt="logo"
            height="50px"
            src="ninja.svg"
          />,
          "iconElementRight": <FlatButton
            disabled={false}
            fullWidth={false}
            href="/"
            label="Back"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          />,
          "onTitleClick": [Function],
          "showMenuIconButton": true,
          "title": "Clipboard.ninja",
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
